onnx/defs/schema.h

```diff
@@ -209,10 +210,25 @@ class OpSchema final {
   OpSchema& SetSupportLevel(SupportType supportType);
 
   // Functions to do documentation for the operator schema.
-  OpSchema& SetDoc(std::string doc);
+  // This may be disabled to save memory.
+  OpSchema& SetDoc(const char* doc) {
+#ifndef ONNX_NO_DOC_STRINGS
```

Comment:
Can you separate the doc string stripping change to anther PR? I prefer not to let these two discussions don't slow down each other. Also according to our previous discussion in #623, removing doc strings really doesn't save you too much (~50kb **uncompressed** size, <10kb **compressed**)

Reply:
I see a similar gain - 56 KB.  I also have to make the strings static and avoid consuming onnx.lib with whole-archive (which is now possible with this change).  I think this is worthwhile for such an isolated change.  Savings like this one add up.

Reply:
This applies to your question below about custom schema.  They may also implement ISchemaRegistry.

Reply:
What's the total size in your measurement? If it's the same as in the previous PR, the total size is around 8MB, which means 50 KB is really not worth adding this complexity into the code base (note as you have found this not only needs to add those macros, but also require moving couple strings and changing some function signature in the OpSchema class, and in some code you will need to add guards preventing accessing the doc strings).
In any case, I feel this really should be a separate discussion, could you split it out into a separate PR.

Reply:
The baseline size was around 2.5 MB.  In the context of a larger product this isn't as relevant.  Components of different sizes sharing a common budget should be equally judicious with size deltas, and this is an easy win.   It also reduces heap usage, which is multiplied across every single process loading the binary.

Doc strings are optional, so guards aren't necessary.  The string movements were necessary for different reasons - the GCC bug workaround.  The signature changes also also needed for other reasons - minimizing static STL string initialization and avoiding instruction bloat in the macro-defined functions.